### PR TITLE
OH2-388 | Add methods to mutate exam with rows

### PR DESCRIPTION
**Pull request by SteveGT96**
See [OH2-388](https://openhospital.atlassian.net/browse/OH2-388)

[OH2-388]: https://openhospital.atlassian.net/browse/OH2-388?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/isf/exa/manager/ExamBrowsingManager.java [hunk 1]
```diff
@@ -35,25 +35,20 @@
...
 
 /**
- * Class that provides gui separation from database operations and gives some
- * useful logic manipulations of the dinamic data (memory)
- *
- * @author bob
- * 19-dec-2005
- * 14-jan-2006
+ * Class that provides gui separation from database operations and gives some useful logic manipulations of the dinamic data (memory)
+ * @author bob 19-dec-2005 14-jan-2006
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/exa/manager/ExamBrowsingManager.java]: Author and revision date information is slowly being removed as git history is sufficient

File: src/main/java/org/isf/exa/manager/ExamBrowsingManager.java [hunk 2]
```diff
@@ -35,25 +35,20 @@
 import org.springframework.stereotype.Component;
 
 /**
- * Class that provides gui separation from database operations and gives some
- * useful logic manipulations of the dinamic data (memory)
- *
- * @author bob
- * 19-dec-2005
- * 14-jan-2006
+ * Class that provides gui separation from database operations and gives some useful logic manipulations of the dinamic data (memory)
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/exa/manager/ExamBrowsingManager.java]:
```suggestion
 * Class that provides gui separation from database operations and gives some useful logic manipulations of the dynamic data (memory)
```

File: src/main/java/org/isf/exa/manager/ExamBrowsingManager.java [hunk 3]
```diff
@@ -131,9 +119,30 @@ public boolean isKeyPresent(Exam exam) throws OHServiceException {
 		return ioOperations.isKeyPresent(exam);
 	}
 
+	/**
+	 * Insert a new {@link Exam} in the DB with exam rows.
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/exa/manager/ExamBrowsingManager.java]: I personally find "in the DB" as being redundant; where else would it be inserted?    There are other examples in this PR like this too.

Comment (SteveGT96) [reply src/main/java/org/isf/exa/manager/ExamBrowsingManager.java]: OK ! Let me fix it !

File: src/main/java/org/isf/exa/manager/ExamBrowsingManager.java [hunk 4]
```diff
@@ -35,25 +35,23 @@
...
 
 /**
- * Class that provides gui separation from database operations and gives some
- * useful logic manipulations of the dinamic data (memory)
- *
+ * Class that provides gui separation from database operations and gives some useful logic
+ * manipulations of the dynamic data (memory)
  * @author bob
  * 19-dec-2005
  * 14-jan-2006
```
Comment (mwithi) [inline src/main/java/org/isf/exa/manager/ExamBrowsingManager.java]: Please remove these:

```suggestion
```

Comment (SteveGT96) [reply src/main/java/org/isf/exa/manager/ExamBrowsingManager.java]: Done !

File: src/main/java/org/isf/exa/service/ExamIoOperations.java [hunk 5]
```diff
@@ -91,19 +92,67 @@ public List<ExamType> getExamType() throws OHServiceException {
...
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public Exam update(Exam payload, List<String> rows) throws OHServiceException {
+		Exam oldExam = findByCode(payload.getCode());
+		Exam exam = repository.save(payload);
+		List<ExamRow> examRows = rowRepository.findAllByExam_CodeOrderByDescription(exam.getCode());
+		if (exam.getProcedure() == 3) {
+			if (oldExam.getProcedure() != 3) {
+				rowRepository.deleteAll(examRows);
```
Comment (mwithi) [inline src/main/java/org/isf/exa/service/ExamIoOperations.java]: Maybe the two if could be combined:

```suggestion
		if (exam.getProcedure() == 3 && oldExam.getProcedure() != 3) {
				rowRepository.deleteAll(examRows);
```

Comment (SteveGT96) [reply src/main/java/org/isf/exa/service/ExamIoOperations.java]: `
if (exam.getProcedure() == 3) {
	if (oldExam.getProcedure() != 3) {
		rowRepository.deleteAll(examRows);
	}
	return exam;
}
`

I want to return the saved exam when its procedure is 3(no extra comptation). In that case, I also want to do delete exam rows if old exam procedure was not 3.

Doing like your're saying doesn't solve that.
Am I wrong ?

Comment (mwithi) [reply src/main/java/org/isf/exa/service/ExamIoOperations.java]: so you could put the whole extra computation in a else block and having only a return statement at the end (more readable)

Comment (SteveGT96) [reply src/main/java/org/isf/exa/service/ExamIoOperations.java]: Oh, I see, that works !

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**